File: backend/app/models/risk.py
import uuid

from sqlalchemy import Column, DateTime, ForeignKey, Integer, String, func
from sqlalchemy.orm import relationship

from app.db.base_class import Base
from app.db.custom_types import GUID, EncryptedString


class UserRiskProfile(Base):
    __tablename__ = "user_risk_profiles"

    id = Column(GUID, primary_key=True, default=uuid.uuid4)
    user_id = Column(GUID, ForeignKey("users.id"), unique=True, nullable=False)
    answers = Column(EncryptedString, nullable=False)
    score = Column(Integer, nullable=True)
    risk_category = Column(String, nullable=True)
    updated_at = Column(
        DateTime,
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )

    user = relationship("User", backref="risk_profile")

File: frontend/src/types/risk.ts
export interface UserRiskProfile {
    id: string;
    user_id: string;
    answers: Record<string, string>;
    score: number | null;
    risk_category: string | null;
    updated_at: string;
}

export interface UserRiskProfileCreate {
    answers: Record<string, string>;
}
